src/lib/shapelib/ShapeComponent.tsx

```diff
@@ -1,16 +1,68 @@
-import React, { useEffect, useRef } from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { drawShape, ShapeProps } from './core';
 
-export const ShapeComponent: React.FC<ShapeProps & React.HTMLAttributes<HTMLElement>> = (props) => {
-  const refToDiv = useRef(null);
+export interface ShapeComponentProps extends ShapeProps { 
+  className?: string;
+  style?: React.CSSProperties;
+  debug?: boolean;
+}
+
+const defaultShapeStyle: React.CSSProperties = {
```

Comment:
I really like the idea of using CSS syntax for styling. However we should take care here in distinguishing between the style of the svg elements and the style of the containing div.